cmd/admin/devices/devices.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*/
+Package main of cmd/admin/device is a command line utility that allows adding
+and removing devices from the device inventory.
+*/
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/onosproject/onos-config/pkg/certs"
+	"github.com/onosproject/onos-config/pkg/northbound"
+	devices "github.com/onosproject/onos-config/pkg/northbound/proto"
+	"io"
+	"log"
+	"strings"
+)
+
+func main() {
+	address := flag.String("address", ":5150", "address to which to send requests e.g. localhost:5150")
+	keyPath := flag.String("keyPath", certs.Client1Key, "path to client private key")
+	certPath := flag.String("certPath", certs.Client1Crt, "path to client certificate")
+
+	addDevice := flag.String("addDevice", "", "protobuf encoding of device to add")
+	removeID := flag.String("removeID", "", "id of device to remove")
+	verbose := flag.Bool("verbose", false, "show all device attributes")
+
+	flag.Parse()
+
+	opts, err := certs.HandleCertArgs(keyPath, certPath)
+	if err != nil {
+		log.Fatal("Error loading cert", err)
+	}
+
+	conn := northbound.Connect(address, opts...)
+	defer conn.Close()
+
+	client := devices.NewDeviceInventoryServiceClient(conn)
+
+	if addDevice != nil && len(*addDevice) > 0 {
+		addNewDevice(client, *addDevice)
+
+	} else if removeID != nil && len(*removeID) > 0 {
+		removeDevice(client, *removeID)
+
+	} else {
+		listDevices(client, *verbose)
+	}
+}
+
+func addNewDevice(client devices.DeviceInventoryServiceClient, deviceProto string) {
+	deviceInfo := &devices.DeviceInfo{}
+	if err := proto.UnmarshalText(deviceProto, deviceInfo); err == nil {
+		fmt.Printf("Adding device %s\n", deviceInfo.Id)
```

Comment:
should this be log ?

Reply:
For server-side, everything should just log. For the client commands though, as we discussed, the fmt.* should be used for plain (undecorated) command output and we can continue to use log for error messages.